native/src/main/java/io/ballerina/stdlib/http/transport/contractimpl/sender/channel/TargetChannel.java

```diff
@@ -82,8 +83,11 @@ public void configTargetHandler(HttpCarbonMessage httpCarbonMessage, HttpRespons
         targetHandler.setTargetChannel(this);
     }
 
-    public BackPressureHandler getBackPressureHandler() {
-        return Util.getBackPressureHandler(targetHandler.getContext());
+    public BackPressureObservable getBackPressureObservable() {
+        if (Util.getBackPressureHandler(targetHandler.getContext()) != null) {
+            return Util.getBackPressureHandler(targetHandler.getContext()).getBackPressureObservable();
+        }
+        return null;
```

Comment:
Instead of returning `null`, I think we should throw an exception.

Reply:
In this part, we check whether the `backPressureHandler` is empty or not to extract the `backPressureObservable`. Since, this is used for HTTP1.1 and HTTP2, we will not have a `backPressureHandler` for HTTP2 (For HTTP2, we get the `backPressureObservable` from `outboundMsgHolder`). That's why we return `null`, and we only update `backPressureHandler` in the `DefaultHttpResponseFuture` if it is not null. Throwing an exception in this point causes test failures.

Reply:
Returning `null` from an API is not good. Because by looking at the API one cannot decide all the possible outcomes.  But in this case, we can tolerate it.